### PR TITLE
case : BUGZ 560: disabling dde initialization due to random crashes while closing down…

### PR DESCRIPTION
**Pull request by amerhifi**
https://highfidelity.atlassian.net/browse/BUGZ-560

Disabling DDE facetracker due to random socket closing crashes on MacOS. Based on stack trace  https://share.backtrace.io/api/share/x2npMNWWbooMNRVtx1hkkYB0 it seems pretty random. Since we don't use the facetracker at this point this is the simplest solution Unfortunately it seems that isOpen functionality is not available to test if the socket is open or closed to try to prevent that crash.

Test case:

This is pretty random. When you close the app on macOs you should not have a crash with similar stack trace.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19833/HighFidelity-Beta-PR15703-b1df463.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19833/HighFidelity-Beta-Interface-PR15703-b1df463.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19833/HighFidelity-Beta-PR15703-b1df463.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19833/HighFidelity-Quest-Beta-PR15703-b1df463.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19833/HighFidelity-Beta-PR15703-b1df463.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19833/HighFidelity-Beta-Interface-PR15703-b1df463.exe

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -5176,7 +5182,11 @@ ivec2 Application::getMouse() const {
 FaceTracker* Application::getActiveFaceTracker() {
     auto dde = DependencyManager::get<DdeFaceTracker>();
 
-    return dde->isActive() ? static_cast<FaceTracker*>(dde.data()) : nullptr;
+    if(dde){
```
Comment (hyperlogic) [inline interface/src/Application.cpp]: Our code standard says this should be:

    if (dde) {
        // blah
    } else {
        // blah
    }

https://github.com/highfidelity/hifi/blob/master/CODING_STANDARD.md#341-the-nominal-case-should-be-put-in-the-if-part-and-the-exception-in-the-else-part-of-an-if-statement

File: interface/src/ui/PreferencesDialog.cpp [hunk 2]
```diff
@@ -30,48 +30,47 @@
 void setupPreferences() {
     auto preferences = DependencyManager::get<Preferences>();
     auto myAvatar = DependencyManager::get<AvatarManager>()->getMyAvatar();
-    static const QString AVATAR_BASICS { "Avatar Basics" };
+    static const QString AVATAR_BASICS{ "Avatar Basics" };
```
Comment (hyperlogic) [inline interface/src/ui/PreferencesDialog.cpp]: Our coding standard says we actually want the space after AVATAR_BASICS and the open curly brace.

https://github.com/highfidelity/hifi/blob/master/CODING_STANDARD.md#322-initialization-of-member-variables-with-default-values

Comment (amerhifi) [reply interface/src/ui/PreferencesDialog.cpp]: I will be backing out the spacing changes. VS unfortunately reformatted the file and spacings

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19846/HighFidelity-Beta-PR15703-9498e0a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19846/HighFidelity-Beta-Interface-PR15703-9498e0a.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19846/HighFidelity-Beta-PR15703-9498e0a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19846/HighFidelity-Quest-Beta-PR15703-9498e0a.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19846/HighFidelity-Beta-PR15703-9498e0a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19846/HighFidelity-Beta-Interface-PR15703-9498e0a.exe

File: interface/src/Application.cpp [hunk 3]
```diff
@@ -5176,7 +5182,11 @@ ivec2 Application::getMouse() const {
 FaceTracker* Application::getActiveFaceTracker() {
     auto dde = DependencyManager::get<DdeFaceTracker>();
 
-    return dde->isActive() ? static_cast<FaceTracker*>(dde.data()) : nullptr;
+    if(dde && dde->isActive()){
```
Comment (hyperlogic) [inline interface/src/Application.cpp]: space after 'if' and before '{'

https://github.com/highfidelity/hifi/blob/master/CODING_STANDARD.md#415-the-if-else-class-of-statements-should-have-the-following-form

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19852/HighFidelity-Beta-PR15703-f2c6cb7.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19852/HighFidelity-Quest-Beta-PR15703-f2c6cb7.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19852/HighFidelity-Beta-PR15703-f2c6cb7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19852/HighFidelity-Beta-Interface-PR15703-f2c6cb7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19852/HighFidelity-Beta-PR15703-f2c6cb7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19852/HighFidelity-Beta-Interface-PR15703-f2c6cb7.exe

**Pull request merged**